src/main/java/core/basesyntax/figures/FigureSupplier.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.figures;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_NUMBER = 30;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        FigureName figureName = FigureName.values()[random.nextInt(FigureName.values().length)];
+        switch (figureName) {
+            case SQUARE:
+                return new Square(colorSupplier.getRandomColor(), random.nextInt(MAX_NUMBER));
```

Comment:
It's better to have a separate method for each figure creation and configuration, like getRandomSquare() and so on